oc_mirror.sh

```diff
@@ -0,0 +1,71 @@
+#!/usr/bin/env bash
+set -ex
+
+# Utility functions to create a local mirror registry using 'mirror-registry' and mirror
+# a release using 'oc mirror'
+
+function add_auth_to_pull_secret() {
+ 
+   quay_auths=`echo -n "${QUAY_REGISTRY_USER}:${QUAY_REGISTRY_PASS}" | base64 -w0`
+
+   tmpauthfile=$(mktemp --tmpdir "quayauth--XXXXXXXXXX")
+   _tmpfiles="$_tmpfiles $tmpauthfile"
+   tmppullsecret=$(mktemp --tmpdir "pullsecret--XXXXXXXXXX")
+   _tmpfiles="$_tmpfiles $tmppullsecret"
+
+
+   cat > "${tmpauthfile}" << EOF
+{
+  "auths": {
+    "${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}": {
+      "auth": "$quay_auths"
+    }
+  }
+}
+EOF
+
+   jq -s '.[0] * .[1]' ${tmpauthfile} ${PULL_SECRET_FILE} > ${tmppullsecret}
+   cp ${tmppullsecret} ${PULL_SECRET_FILE}
+
+   # Note that this must also be manually added to .docker/config.json, we don't currently do this in dev-scripts to avoid writing
+   # to user directory.
+
+}
+
+function setup_quay_mirror_registry() {
+
+   mkdir -p ${WORKING_DIR}/oc-mirror
+   pushd ${WORKING_DIR}/mirror-registry
+   sudo ./mirror-registry install --quayHostname virthost.ostest.test.metalkube.org --quayRoot ${WORKING_DIR}/oc-mirror/ --initUser ${QUAY_REGISTRY_USER} --initPassword ${QUAY_REGISTRY_PASS} --sslCheckSkip -v
+
+   add_auth_to_pull_secret
+   popd
+}
+
+function publish_mirror_registry() {
+
+   # Create imageset containing the local URL and the OCP release to mirror
+   tmpimageset=$(mktemp --tmpdir "imageset--XXXXXXXXXX")
+   _tmpfiles="$_tmpfiles $tmpimageset"
+
+   cat > "${tmpimageset}" << EOF
+apiVersion: mirror.openshift.io/v1alpha1
+kind: ImageSetConfiguration
+archiveSize: 4
+storageConfig:
+  registry:
+    imageURL: ${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT}/origin:latest
+    skipTLS: true
+mirror:
+  ocp:
+    channels:
+    - name: candidate-${OPENSHIFT_RELEASE_STREAM}
+  additionalImages:
+  - name: registry.redhat.io/ubi8/ubi:latest
+EOF
+
+   # TODO - print message on failure that password for must be added to .docker/config.json
+   # The failure may have an error such as "UNAUTHORIZED: access to the requested resource is not authorized"
+   oc mirror --dest-skip-tls --config ${tmpimageset} docker://${LOCAL_REGISTRY_DNS_NAME}:${LOCAL_REGISTRY_PORT} --log-level debug
```

Comment:
Again this might be an oc version issue, as I see it as an option with 4.12
```
$ oc mirror --help | grep log-level
      --log-level string    Log level (e.g. "debug | info | warn | error") (default "info")

```

Reply:
It turns out I was using an older version of oc-mirror which is why I didn't see that the options changed. I've updated to add a oc_mirror_cleanup.sh to remove oc-mirror and ensure the latest will be used on the next run.